### ISSUE TITLE
move folder and subfolder menu to mda_folder.ui

### ISSUE THREAD

**Issue by rodolakis** [BCDA-APS/mdaviz#132]
`mda_folder.ui` would become:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Form</class>
 <widget class="QWidget" name="Form">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1071</width>
    <height>466</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_3">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>6</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QSplitter" name="hsplitter">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <widget class="QSplitter" name="vsplitter">
      <property name="orientation">
       <enum>Qt::Vertical</enum>
      </property>
      <widget class="QGroupBox" name="folder_groupbox">
       <property name="title">
        <string/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <layout class="QGridLayout" name="gridLayout" columnstretch="1,5,1">
          <property name="leftMargin">
           <number>0</number>
          </property>
          <property name="topMargin">
           <number>0</number>
          </property>
          <property name="rightMargin">
           <number>0</number>
          </property>
          <property name="bottomMargin">
           <number>6</number>
          </property>
          <item row="0" column="0">
           <widget class="QLabel" name="folder_label">
            <property name="toolTip">
             <string>Selected parent folder</string>
            </property>
            <property name="text">
             <string>Full Path:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_2">
            <item>
             <widget class="QPushButton" name="open">
              <property name="toolTip">
               <string>Open a new folder</string>
              </property>
              <property name="icon">
               <iconset>
                <normaloff>open.png</normaloff>open.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>14</width>
                <height>14</height>
               </size>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QPushButton" name="refresh_2">
              <property name="toolTip">
               <string>Refresh folder content</string>
              </property>
              <property name="icon">
               <iconset>
                <normaloff>refresh.png</normaloff>refresh.png</iconset>
              </property>
              <property name="iconSize">
               <size>
                <width>14</width>
                <height>14</height>
               </size>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="subfolder_2">
            <property name="toolTip">
             <string>Select folder or subfolder from full path</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="subfolder_label_2">
            <property name="toolTip">
             <string>Select folder or subfolder from full path</string>
            </property>
            <property name="text">
             <string>Subfolder(s):</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="folder">
            <property name="toolTip">
             <string>Selected parent folder</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <layout class="QHBoxLayout" name="horizontalLayout_4">
            <item>
             <spacer name="horizontalSpacer">
              <property name="orientation">
               <enum>Qt::Horizontal</enum>
              </property>
              <property name="sizeType">
               <enum>QSizePolicy::Minimum</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>10</width>
                <height>20</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <widget class="QLabel" name="info">
              <property name="text">
               <string>Folder info</string>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </widget>
      <widget class="QGroupBox" name="mda_groupbox">
       <property name="title">
        <string/>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <property name="spacing">
         <number>0</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
       </layout>
      </widget>
     </widget>
     <widget class="QGroupBox" name="viz_groupbox">
      <property name="title">
       <string/>
      </property>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="spacing">
        <number>0</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>

```

**Comment by rodolakis** [comment]
It means the `folder`, `subfolder` and `info` attribute now become `self.mvc_folder.folder` etc...

In `MainWindow.setup()`, `self.mvc_folder = None`, so connecting the signals for `self.mvc_folder.folder` and `self.mvc_folder.subfolder` becomes tricky. Maybe it is just about using a `try...except` but then does it establish the connection? Or does that mean I just need to re-establish the connection when I set the `self.mvc_folder = MDA_MVC(self)`?
It may not be too hard but was more complicated that I intended on a Saturday evening...
